mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -2333,6 +2312,41 @@ def NVVM_Breakpoint : NVVM_Op<"breakpoint"> {
   let assemblyFormat = "attr-dict";
 }
 
+def NVVM_Trap : NVVM_Op<"trap"> {
```

Comment:
Do you mean `llvm.intr.trap`? Right, we could use that. I don't have a strong opinion.. But it might be easy to codegen. 

The NVVM dialect also has `nvvm.breakpoint`, which is the same as `llvm.intr.debugtrap`.

If we want to use LLVM's version, should we also remove `nvvm.breakpoint`?

Reply:
@durga4github what do you think? @schwarzschild-radius has a good point.
Should we just use the llvm dialect's OPs or have our own in NVVM dialect? I don't have a strong opinion.

Reply:
What is the lowering to LLVM for this MLIR op? Is there a LLVM intrinsic other than llvm.trap for NVVM? (and what's the difference? At minima it should be documented).

Reply:
> If you're lowering to llvm.trap then I'm not sure why not have just llvm.trap in MLIR?
+1

> If we want to use LLVM's version, should we also remove nvvm.breakpoint?
That's a good point, I think we can. nvvm.breakpoint was added to model the PTX brkpt instruction. Since it lowers to debugtrap (more like an alias), it is not always to trivial for the user to use llvm.debugtrap Op. Whereas with `trap`, it maps one-to-one with the PTX instruction

I am wondering if we should have a table of mapping from LLVM/NVVM dialect to PTX to help the users

Reply:
I originally considered having separate Ops for the "PTX Misc Instructions" section.

From Pradeep's latest comment: it seems breakpoint lowering/intrinsic mapping seems not as straightforward as trap.
1)
So, for the trap Op, I am inclined to use llvm.trap itself.
2)
And then, for the, nvvm.breakpoint Op: Add some documentation on why we have a separate Op here (instead of using the llvm dialect Op).

@schwarzschild-radius , @grypp Does it look reasonable?

Reply:
Sounds good to me!

Reply:
Sure sounds good to me as well! I'll close the PR. 


By the way, trap and debugpoint both generates the same PTX instruction. Is it expected?